src/auth/auth.ts

```diff
@@ -351,6 +352,49 @@ export class BaseAuth<T extends AbstractAuthRequestHandler> implements BaseAuthI
    * @return {Promise<UserRecord>} A promise that resolves with the modified user record.
    */
   public updateUser(uid: string, properties: UpdateRequest): Promise<UserRecord> {
+    // Although we don't really advertise it, we want to also handle linking of
+    // non-federated idps with this call. So if we detect one of them, we'll
+    // adjust the properties parameter appropriately. This *does* imply that a
+    // conflict could arise, e.g. if the user provides a phoneNumber property,
+    // but also provides a providerToLink with a 'phone' provider id. In that
+    // case, we'll throw an error.
+    properties = deepCopy(properties);
+    if (validator.isNonNullObject(properties) && typeof properties.providerToLink !== 'undefined') {
```

Comment:
```
if (properties?.providerToLink) {

}
```

Reply:
Done. I've also applied it to the providerToUnlink variable a few lines below.